File: app/src/main/java/dev/ragnarok/fenrir/media/voice/VoicePlayerFactory.java
package dev.ragnarok.fenrir.media.voice;

import android.content.Context;

import androidx.annotation.NonNull;

import dev.ragnarok.fenrir.model.ProxyConfig;
import dev.ragnarok.fenrir.settings.IProxySettings;
import dev.ragnarok.fenrir.settings.ISettings;

public class VoicePlayerFactory implements IVoicePlayerFactory {

    private final Context app;
    private final IProxySettings proxySettings;
    private final boolean NotSensered;

    public VoicePlayerFactory(Context context, IProxySettings proxySettings, ISettings.IOtherSettings otherSettings) {
        app = context.getApplicationContext();
        this.proxySettings = proxySettings;
        NotSensered = otherSettings.isDisable_sensored_voice();
    }

    @NonNull
    @Override
    public IVoicePlayer createPlayer() {
        ProxyConfig config = proxySettings.getActiveProxy();
        return NotSensered ? new ExoVoicePlayer(app, config) : new ExoVoicePlayerSensored(app, config);
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/media/voice/VoicePlayerFactory.kt
package dev.ragnarok.fenrir.media.voice

import android.content.Context
import dev.ragnarok.fenrir.settings.IProxySettings
import dev.ragnarok.fenrir.settings.ISettings

class VoicePlayerFactory(
    context: Context,
    private val proxySettings: IProxySettings,
    mainSettings: ISettings.IMainSettings
) : IVoicePlayerFactory {
    private val app: Context = context.applicationContext
    private val NotSensered: Boolean = mainSettings.isDisable_sensored_voice
    override fun createPlayer(): IVoicePlayer {
        val config = proxySettings.activeProxy
        return if (NotSensered) ExoVoicePlayer(app, config) else ExoVoicePlayerSensored(app, config)
    }

}